### PR TITLE
I539 - request password reset form

### PR DESCRIPTION
**Pull request by littledice**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-539

### CODE REVIEW

File: app/src/main/admin/components/AdminRouter.tsx [hunk 1]
```diff
@@ -30,6 +31,7 @@ export class AdminRouter extends Router<RouterProps> {
             map('/users/:username', ViewUserDetailsPage);
             map('*', AdminNoRouteFoundPage);
         } else {
+            map('/forgotten-password', AdminForgottenPasswordPage);
             map('*', AdminLoginPage);
         }
     }
```
Comment (MartinEden) [inline app/src/main/admin/components/AdminRouter.tsx]: This URL should be mapped regardless of whether the user is logged in, perhaps?

File: app/src/main/contrib/components/ContribRouter.ts [hunk 2]
```diff
@@ -34,6 +35,7 @@ export class ContribRouter extends Router<RoutingProperties> {
                 map("*", LoadingPage);
             }
         } else {
+            map('/forgotten-password', ContribForgottenPasswordPage);
             map('*', ContribLoginPage);
         }
     }
```
Comment (MartinEden) [inline app/src/main/contrib/components/ContribRouter.ts]: Ditto: 

> This URL should be mapped regardless of whether the user is logged in, perhaps?

File: app/src/main/report/components/ReportingRouter.tsx [hunk 3]
```diff
@@ -24,6 +25,7 @@ export class ReportingRouter extends Router<RouterProps> {
             map('/:report/:version', VersionInfoPage);
             map('*', ReportingNoRouteFoundPage);
         } else {
+            map('/forgotten-password', ReportingForgottenPasswordPage);
             map('*', ReportingLoginPage);
         }
     }
```
Comment (MartinEden) [inline app/src/main/report/components/ReportingRouter.tsx]: Ditto:

> This URL should be mapped regardless of whether the user is logged in, perhaps?

File: app/src/main/shared/actions/AuthActions.ts [hunk 4]
```diff
@@ -43,6 +44,10 @@ class AuthActions extends AbstractActions implements Actions {
     logOut(): boolean {
         return true;
     }
+
+    sendPasswordResetEmail() : void {
+
+    }
```
Comment (MartinEden) [inline app/src/main/shared/actions/AuthActions.ts]: Don't actions needs to return something (e.g. `true`) in order for Alt to dispatch them?

**Comment by littledice** [comment]
All changes made except 2. above which is a nice to have
I've opened a ticket: https://vimc.myjetbrains.com/youtrack/issue/VIMC-552

**Pull request merged**